common/logger/logger.ts

```diff
@@ -0,0 +1,74 @@
+import { isCommandArg } from '../util/basic';
+import { configure, addLayout, getLogger as getlog4jsLogger, Logger as Log4jsLogger } from 'log4js';
+
+addLayout('json', function () {
+    return function (logEvent) {
+        const message = logEvent.data.shift();
+        const data = logEvent.data.shift();
+        return JSON.stringify({ ...logEvent, message, data });
+    };
+});
+
+let appenders = ['out'];
+if (process.env.LOG_FORMAT === 'json') {
+    appenders = ['outJson'];
+}
+
+configure({
+    appenders: {
+        out: { type: 'stdout', layout: { type: 'coloured' } },
+        outJson: { type: 'stdout', layout: { type: 'json' } },
+    },
+    categories: {
+        default: {
+            appenders,
+            level: isCommandArg('--debug') ? 'debug' : 'info',
+        },
+    },
+});
+
+export type LogData = { [key: string]: any };
+
+export class Logger {
```

Comment:
This is a wrapper class to abstract the logging library used. It enables us to use other components in the future.